assembl/static2/js/app/components/administration/saveButton.jsx

```diff
@@ -40,57 +43,90 @@ const createVariablesForMutation = (thematic) => {
   };
 };
 
-const SaveButton = ({ createThematic, deleteThematic, enabled, refetchThematics, thematics, updateThematic }) => {
+const SaveButton = ({ i18n, createThematic, deleteThematic, enabled, refetchThematics, thematics, updateThematic, client, updateDiscussionPreference, preferences, languagePreferenceHasChanged, resetLanguagePreferenceChanged}) => {
   const saveAction = () => {
     displayAlert('success', `${I18n.t('loading.wait')}...`);
     const promisesArray = [];
-    thematics.forEach((thematic) => {
-      if (thematic.isNew && !thematic.toDelete) {
-        // create thematic
-        const payload = {
-          variables: createVariablesForMutation(thematic)
-        };
-        const p1 = () => {
-          return createThematic(payload);
-        };
-        promisesArray.push(p1);
-      } else if (thematic.toDelete && !thematic.isNew) {
-        // delete thematic
-        const payload = {
-          variables: {
-            thematicId: thematic.id
-          }
-        };
-        const p3 = () => {
-          return deleteThematic(payload);
-        };
-        promisesArray.push(p3);
-      } else {
-        // update thematic
-        const variables = createVariablesForMutation(thematic);
-        variables.id = thematic.id;
-        const payload = {
-          variables: variables
-        };
-        const p2 = () => {
-          return updateThematic(payload);
-        };
-        promisesArray.push(p2);
+    //Compare the redux state change to the saved data in apollo
+
+    if (languagePreferenceHasChanged) {
+      //Save and update the apolloStore
+      const payload = {
+        variables: {
+          languages: preferences
+        },
+        update: (storeProxy, { data: { updateDiscussionPreference: { preferences: { languages } } } }) => {
+          //Update the apollo cache
+          const query = storeProxy.readQuery({
+            query: getDiscussionPreferenceLanguage,
+            variables: { inLocale: i18n.locale }
+          });
+          const newData = {...query};
+          newData.discussionPreferences.languages = languages; 
+          storeProxy.writeQuery({
```

Comment:
I don't understand everything here, but when you call the first mutation, does it return new values of languages? If yes, you would not need to send another query afterwards

Reply:
That is certainly one way of doing it. When I send the mutation on the wire, upon a success, I can ask for a `refetchQueries` and ask apollo to refetch the `discussionPreferences`, but it is wasteful (another request on the wire). Instead, I take the response of the mutation, and update the apollo cache directly. Components who subscribed to the cache will also see the update.